Update macro arguments to specify rules_ios modified arguments

luispadron opened the pull request:
A follow-up to #670 where we talked about moving the rules_ios modified kwargs to explicit arguments in the macros. Additionally, generates missing docs & updates outdated documentation.

luispadron commented on rules/app.bzl:
```diff
@@ -67,7 +83,6 @@ def ios_application(name, apple_library = apple_library, infoplists_by_build_set
     )
 
     application_kwargs["launch_storyboard"] = application_kwargs.pop("launch_storyboard", library.launch_screen_storyboard_name)
-    application_kwargs["families"] = application_kwargs.pop("families", ["iphone", "ipad"])
```
Not sure why we decided to default this, but agree it's probably best to not default here and instead make sure the user provides it.

This will be a breaking change though so not sure if its worth it? Let me know

thiagohmcruz replied:
I also don't know why we default-ed here, since this is required on the `rules_apple` side I think it makes sense to default to empty and allow consumers to configure this properly

I'll tag @justinseanmartin and @congt here since they might have context on older versions of this file and raise any concerns with this breaking change in favour of a more reasonable default

That said, I think it's a small change on users BUILD files, so I'd only consider not doing it if there's a major concern here

thiagohmcruz replied:
I'm ok with not reverting it then 👍, +1 to making it visible in the docs

thiagohmcruz replied:
@mattrobmattrob so just to confirm the preference would be to keep the two families as default everywhere or only for apps?

mattrobmattrob replied:
I suppose I'd prefer the default (assuming we keep it) is the same anywhere it's applicable as a param to the underlying rules.

luispadron replied:
I'll update to move that to the macro args so its documented in all places it was previously used 👍🏼

thiagohmcruz commented on rules/framework.bzl:
```diff
@@ -40,27 +40,48 @@ _APPLE_FRAMEWORK_PACKAGING_KWARGS = [
-    default_infoplists = kwargs.pop("infoplists", [])
-    infoplists = None
-    if len(infoplists_by_build_setting.values()) > 0 or len(default_infoplists) > 0:
+    merged_infoplists = None
```
`merged_infoplists` is not being set anywhere IIUC?

luispadron replied:
```
    apple_framework_packaging(
        name = name,
        framework_name = library.namespace,
        infoplists = merged_infoplists,
```

Gets set on the framework

thiagohmcruz replied:
yep but if I followed the code paths correctly it's set to `None` above and then passed as `None` to the fmw right? I don't see it receive any non-`None` value anywhere

luispadron replied:
🤦🏼 ah right, i've updated it to get set as the result of the `process_infoplist`!

thiagohmcruz reviewed (approved):
I think it's a good idea to standardize these, also thx for adding the new docs! I'd get feedback from others too re the minor breaking change here but AFAICT it's not major.

mattrobmattrob commented on rules/app.bzl:
```diff
@@ -39,20 +39,36 @@ _IOS_APPLICATION_KWARGS = [
     Args:
         name: The name of the iOS application.
         apple_library: The macro used to package sources into a library.
+        infoplists: A list of Info.plist files to be merged into the app clip.
```
```suggestion
        infoplists: A list of Info.plist files to be merged into the iOS app.
```

mattrobmattrob commented on rules/extension.bzl:
```diff
@@ -14,22 +20,29 @@ def ios_extension(name, infoplists_by_build_setting = {}, **kwargs):
 
     Args:
         name: The name of the iOS extension.
+        infoplists: A list of Info.plist files to be merged into the app clip.
```
```suggestion
        infoplists: A list of Info.plist files to be merged into the extension.
```

mattrobmattrob commented on rules/extension.bzl:
```diff
@@ -14,22 +20,29 @@ def ios_extension(name, infoplists_by_build_setting = {}, **kwargs):
 
                                      If '//conditions:default' is not set the value in 'infoplists'
                                      is set as default.
+        xcconfig: A dictionary of xcconfigs to be applied to the app clip by default.
```
```suggestion
        xcconfig: A dictionary of xcconfigs to be applied to the extension by default.
```

mattrobmattrob commented on rules/framework.bzl:
```diff
@@ -40,27 +40,48 @@ _APPLE_FRAMEWORK_PACKAGING_KWARGS = [
     Args:
         name: The name of the framework.
         apple_library: The macro used to package sources into a library.
+        infoplists: A list of Info.plist files to be merged into the app clip.
```
```suggestion
        infoplists: A list of Info.plist files to be merged into the framework.
```

mattrobmattrob commented on rules/framework.bzl:
```diff
@@ -40,27 +40,48 @@ _APPLE_FRAMEWORK_PACKAGING_KWARGS = [
+
+                                     If '//conditions:default' is not set the value in 'infoplists'
+                                     is set as default.
+        xcconfig: A dictionary of xcconfigs to be applied to the app clip by default.
```
```suggestion
        xcconfig: A dictionary of xcconfigs to be applied to the framework by default.
```

mattrobmattrob reviewed (approved):
I'd prefer we leave the default `families` value but make it show up in the docs.

luispadron commented:
@mattrobmattrob @thiagohmcruz thanks for the reviews! I've updated this to default `families` to `iphone, ipad` and added them to the macros args so they're documented.
